core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -513,6 +513,12 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
   final def either(implicit ev: CanFail[E]): URIO[R, Either[E, A]] =
     self.foldM(ZIO.succeedLeft, ZIO.succeedRight)
 
+  /**
+   * Maps this effect to the default exit codes.
+   */
+  final def exitCode: URIO[R, ExitCode] =
+    self.foldCauseM(cause => UIO(println(cause.prettyPrint)) as ExitCode.failure, _ => ZIO.succeedNow(ExitCode.success))
```

Comment:
Let’s use putStrLn instead of println here.

Reply:
@iravid I took this combinator from #2153, I suspect John suggested `println` to avoid adding a dependency on Console at the end of your program. OTOH, App uses `ZEnv` anyway so even if you write `program.provideSomeLayer(ZEnv.live ++ customLayers).exitCode` it won't crash as Console would be its only missing dependency which the default runtime will provide.

I will update in a few minutes (cc: @jdegoes)

Reply:
@iravid done